### PR TITLE
Improve development experience for Calamari on linux

### PR DESCRIPTION
**Pull request by TomPeters**
Added the ability to open and build the solution on a linux or macOS development environment.

### CODE REVIEW

**Comment by jburger** [comment]
Re: unloading a project from a solution - I used the approach of preventing it from building instead. Would that work for you?

```
<Target Condition="$([MSBuild]::IsOSUnixLike())" Name="Build">
    <Message Text="This build is unsupported on unix like environments"/>
  </Target>
<Import Condition="!$([MSBuild]::IsOSUnixLike())" Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
```

**Comment by TomPeters** [comment]
I'm not quite sure what the right msbuild magic is to do that in this specific csproj

**Comment by TomPeters** [comment]
Another interesting warning I'm getting on all the projects:

```
11:24 am	Project 'Calamari.Terraform' load finished with warnings
						The "ResolvePackageDependencies" task failed unexpectedly.
						NuGet.Packaging.Core.PackagingException: Unable to find fallback package folder 'C:\Program Files\dotnet\sdk\NuGetFallbackFolder'.
						at NuGet.Packaging.FallbackPackagePathResolver..ctor(String userPackageFolder, IEnumerable`1 fallbackPackageFolders)
						at Microsoft.NET.Build.Tasks.NuGetPackageResolver.CreateResolver(LockFile lockFile, String projectPath)
						at Microsoft.NET.Build.Tasks.ResolvePackageDependencies.get_PackageResolver()
						at Microsoft.NET.Build.Tasks.ResolvePackageDependencies.ResolvePackagePath(LockFileLibrary package)
						at Microsoft.NET.Build.Tasks.ResolvePackageDependencies.GetPackageAndFileDefinitions()
						at Microsoft.NET.Build.Tasks.ResolvePackageDependencies.ExecuteCore()
						at Microsoft.NET.Build.Tasks.TaskBase.Execute()
						at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
						at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost... (show balloon)
```

I don't know why it would be looking for `C:\Program Files\dotnet\sdk\NuGetFallbackFolder`...

**Comment by jburger** [comment]
Yeah thats interesting. Several workarounds listed here for that one https://github.com/dotnet/cli/issues/8806

File: source/Calamari.Azure/Calamari.Azure.csproj [hunk 1]
```diff
@@ -105,4 +106,8 @@
     <Copy SourceFiles="@(FSharpFiles)" DestinationFolder="$(PublishDir)/FSharp/" />
     <Copy SourceFiles="@(ScriptCSFiles)" DestinationFolder="$(PublishDir)/ScriptCS/" />
   </Target>
+  <Target Condition="$([MSBuild]::IsOSUnixLike())" Name="Build">
+    <Message Text="This build is unsupported on unix like environments"/>
+  </Target>
+  <Import Condition="!$([MSBuild]::IsOSUnixLike())" Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
```
Comment (TomPeters) [inline source/Calamari.Azure/Calamari.Azure.csproj]: I managed to get this working by moving from implicit to explicit sdk imports: https://docs.microsoft.com/en-us/visualstudio/msbuild/how-to-use-project-sdk?view=vs-2019 This gave me something that I could then conditionally skip.

Comment (jburger) [reply source/Calamari.Azure/Calamari.Azure.csproj]: Nicely done :+1:

**Pull request merged**